keras/src/ops/numpy_test.py

```diff
@@ -9401,3 +9401,75 @@ def call(self, x):
         model.compile(jit_compile=jit_compile)
 
         model.predict(np.random.randn(1, 8))
+
+
+class NumPyArraySplitTest(testing.TestCase):
+    @parameterized.named_parameters(
+        ("even_split_1D", 10, 2, 0, (5, 5)),
+        ("uneven_split_1D", 10, 3, 0, (4, 3, 3)),
+        ("one_section_1D", 10, 1, 0, (10,)),
+        ("split_by_length_1D", 7, 3, 0, (3, 2, 2)),
+        ("even_split_2D_axis0", (4, 3), 2, 0, ((2, 3), (2, 3))),
+        ("uneven_split_2D_axis0", (4, 3), 3, 0, ((2, 3), (1, 3), (1, 3))),
+        ("uneven_split_2D_axis1", (3, 5), 3, 1, ((3, 2), (3, 2), (3, 1))),
+        ("single_element_array", 1, 1, 0, (1,)),
+        ("split_by_size_zero_length", 0, 2, 0, (0, 0)),
+        ("split_by_size_zero_length_non_empty_split", 1, 2, 0, (1, 0)),
+        ("3D_axis_neg1", (2, 3, 7), 3, -1, ((2, 3, 3), (2, 3, 2), (2, 3, 2))),
+    )
+    def test_array_split_static_shape_and_correctness(
+        self, shape, indices_or_sections, axis, expected_shapes
+    ):
+        if isinstance(shape, int):
+            input_array = np.arange(shape)
+        else:
+            input_array = np.arange(np.prod(shape)).reshape(shape)
```

Comment:
It's actually quite confusion to have ints in the parameters when it's a shape. So remove the `if` here and instead turn the ints to tuples of one value in the parameters:

```
("even_split_1D", (10,), 2, 0, (5, 5)),
```

This way it makes it obvious it's a shape.